server/game/cards/02_SHD/units/ClanChallengers.ts

```diff
@@ -0,0 +1,22 @@
+import AbilityHelper from '../../../AbilityHelper';
+import { NonLeaderUnitCard } from '../../../core/card/NonLeaderUnitCard';
+import { KeywordName } from '../../../core/Constants';
+
+export default class ClanChallengers extends NonLeaderUnitCard {
+    protected override getImplementationId() {
+        return {
+            id: '3487311898',
+            internalName: 'clan-challengers',
+        };
+    }
+
+    public override setupCardAbilities() {
+        this.addConstantAbility({
+            title: 'While this unit is upgraded, it gains Overwhelm',
+            condition: (context) => context.source.upgraded,
```

Comment:
Can we update "upgraded" to be a function (i.e. isUpgraded() or hasUpgrades() or something)?